src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/resource_allocation_manager_impl.cc

```diff
@@ -248,8 +248,11 @@ void ConstructOnRCStatusNotificationParams(
     return [&result_modules](const ModuleUid& module) {
       smart_objects::SmartObject module_data =
           SmartObject(smart_objects::SmartType_Map);
-      auto module_type =
-          StringToEnum<mobile_apis::ModuleType::eType>(module.first);
+      mobile_apis::ModuleType::eType module_type =
+          mobile_apis::ModuleType::INVALID_ENUM;
+      smart_objects::EnumConversionHelper<
+          mobile_apis::ModuleType::eType>::StringToEnum(module.first,
+                                                        &module_type);
```

Comment:
Seems this is the only use of this function, (defined lines 227-233) so it could be removed.

Alternatively, if this convenience function could be useful, you might be able to avoid the warning by replacing
```cpp
EnumType val;
```
with

```cpp
EnumType val = EnumType::INVALID_ENUM;
```